nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/main/java/org/nuxeo/ecm/automation/core/impl/adapters/helper/TypeAdapterHelper.java

```diff
@@ -31,38 +31,42 @@
 
 /**
  * Helper for type adapters
- * 
+ *
  * @since 9.1
  */
 public class TypeAdapterHelper {
 
     /**
-     * Create document reference from its path
+     * Create document reference from an id or a path.
      *
-     * @param value the document path
      * @return the document reference
      */
     public static DocumentRef createDocumentRef(String value) {
-        return value.startsWith("/") ? new PathRef(value) : new IdRef(value);
+        if (value.startsWith("/")) {
+            return new PathRef(value);
+        }
+        // value could be of the form "repositoryName:docId|docPath"
+        // if so, remove the repositoryName and try to fetch the document with the currently opened session
```

Comment:
I think the "and try to fetch the document with the currently opened session" part can be removed, this method just create a DocumentRef and doesn't fetch anything.